pages/blog/getting-the-best-out-of-your-graphql-client-by-using-relay-patterns-through-graphql-code-generator.mdx

```diff
@@ -0,0 +1,212 @@
+import withArticle from '../../ui/blog/article'
+
+export const meta = {
+  title:
+    'Getting the best out of your GraphQL client by using Relay patterns through GraphQL Code Generator',
+  author: 'arda',
+  tags: ['graphql'],
+  date: '2022-08-01',
+  description: 'TBA',
+  image:
+    '/blog-assets/getting-the-best-out-of-your-graphql-client-by-using-relay-patterns-through-graphql-code-generator/cover.jpg'
+}
+
+export default withArticle(meta)
+
+# Getting the best out of your GraphQL client by using Relay patterns through GraphQL Code Generator
+
+There is no doubt that Relay is the most advanced JavaScript GraphQL Client that is out there. However, at the same time, the learning curve and adoption is yet very sparse compared to other popular alternatives such as Apollo Client, Urql, or GraphQL Request.
+
+As a consequence, many people don’t even know about all the benefits and patterns used within (and so far being exclusive to ) the relay-compiler, the GraphQL Code Generator counter-part tailored for Relay applications.
+
+The Guild gathered and worked on projects whose GraphQL client usage spans across all the available solutions today. In our opinion, the most important parts of Relay are the concepts of building and scaling applications. These patterns are actually applicable to any GraphQL Client, you just need the right code generation tool for the job.
+
+Fortunately, we are happy to announce that we now have an opinionated preset for bringing the Relay patterns to your existing project without having to switch over to the Relay eco-system.
+
+## Describe component data needs via GraphQL fragments
+
+Instead of writing one big GraphQL Operation for our whole page and passing that down to the components, start with describing the component's data dependencies through a GraphQL fragment.
+
+```tsx
+import { gql, useFragment, FragmentType } from '@gql'
+
+const Avatar_UserFragment = gql(/* GraphQL */ `
+  fragment Avatar_UserFragment on User {
+    avatarUrl
+  }
+`)
+
+type AvatarProps = {
+  user: FragmentType<typeof Avatar_UserFragment>
+}
+
+export function Avatar(props: AvatarProps) {
+  const user = useFragment(Avatar_UserFragment, props.user)
+  return <CircleImage src={user.avatarUrl} />
+}
+```
+
+## Compose fragments for UI components
```

Comment:
I would move this section below the fragment masking one and keep a single section after about combining fragments at components or top-level.